pandas/core/dtypes/missing.py

```diff
@@ -292,7 +292,7 @@ def _isna_array(values: ArrayLike, inf_as_na: bool = False):
             # "Union[ndarray[Any, Any], ExtensionArraySupportsAnyAll]", variable has
             # type "ndarray[Any, dtype[bool_]]")
             result = values.isna()  # type: ignore[assignment]
-    elif is_string_or_object_np_dtype(values.dtype):
+    elif is_string_or_object_np_dtype(values.dtype) or isinstance(values, np.recarray):
```

Comment:
Hmm appears this function doesn't deal with output formatting? Why is the fixed specifically needed here?

Reply:
I've put it there because when the dtype is a np.recarray, then `result = np.isnan(values)` in `core/dtypes/missing.py` will be executed and np.isnan cannot be evaluated on a np.recarray, it will throw the error which was also described in the issue.

Reply:
This fix still appears like a band-aid fix. pandas doesn't have great support for `np.recarray`s and it doesn't look like `_isna_string_dtype` was specifically meant to deal with them

Reply:
Okay, then I have a look what one can do.

Reply:
@mroeschke 
Since `_isna_string_dtype` is not meant to deal with `np.recarray`, I have introduced a function that deals with that, can you have a look and tell me if this is okay? Thanks.